util/secret/secret.go

```diff
@@ -56,7 +56,12 @@ func Name(cluster string, suffix Purpose) string {
 // ParseSecretName return the cluster name and the suffix Purpose in name is a valid cluster secrets,
 // otherwise it return error.
 func ParseSecretName(name string) (string, Purpose, error) {
-	separatorPos := strings.LastIndex(name, "-")
+	var separatorPos = strings.LastIndex(name, "-")
+
+	if strings.Contains(name, string(UserKubeconfig)) {
+		separatorPos = strings.LastIndexAny(name[:separatorPos], "-")
```

Comment:
Out of curiosity: why use `strings.LastIndexAny` instead of `strings.LastIndex`?